File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        String[] word1Arr = word1.split("");
        String[] word2Arr = word2.split("");
        StringBuilder result = new StringBuilder();

        if(word1Arr.length == word2Arr.length) {
            for(int i=0; i<word1Arr.length; i++) {
                result.append(word1Arr[i]);
                result.append(word2Arr[i]);
            }
        } else if (word1Arr.length > word2Arr.length) {
            for (int i=0; i< word2Arr.length; i++) {
                result.append(word1Arr[i]);
                result.append(word2Arr[i]);
            }

            for(int j=word2Arr.length; j< word1Arr.length; j++) {
                result.append(word1Arr[j]);
            }

        } else if (word2Arr.length > word1Arr.length) {
            for (int i=0; i< word1Arr.length; i++) {
                result.append(word1Arr[i]);
                result.append(word2Arr[i]);
            }

            for(int j= word1Arr.length; j< word2Arr.length; j++) {
                result.append(word2Arr[j]);
            }
        }
        
        return result.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.kt
class Solution {
fun mergeAlternately(word1: String, word2: String): String {
    var result = ""
    var i = 0
    var j = 0
    while (i < word1.length && j < word2.length) {
        result += word1[i++]
        result += word2[j++]
    }
    while (i < word1.length) {
        result += word1[i++]
    }
    while (j < word2.length) {
        result += word2[j++]
    }
    return result
}
}